modules/dhcp_common/isc/configuration_parser.rb

```diff
@@ -427,11 +427,15 @@ def visit_parse_tree_nodes(parse_tree, all_hosts, all_subnets, all_ignored, root
         end
 
         def parse_file(a_path)
+          if defined? @@config_path
+            a_path = File.absolute_path(a_path, @@config_path)
+          end
           File.open(a_path, 'r:ASCII-8BIT') {|f| conf.parse!(f.read, a_path)}
         end
 
         # returns all_subnets, all_hosts, root_group
         def subnets_hosts_and_leases(conf_as_string, filename)
+          @@config_path = File.dirname(filename)
```

Comment:
This is really bad hack, we need to figure it out. Asked Dmitri who wrote the parser, I am not sure how to properly access it.